### ISSUE TITLE
b.Entity is not iterable

### ISSUE THREAD

**Issue by RaithSphere** [PolyLogiX-Studio/Neos.js#66]
Version: 1.6.2

**Describe the bug**
Headless causing crash with the error, no input from user server is idle

/home/raith/test/node_modules/@bombitmanbomb/neosjs/API/CloudXInterface.js:1703
      for (let item of b.Entity) a.Add(new Message(item));
                         ^

TypeError: b.Entity is not iterable
    at /home/raith/test/node_modules/@bombitmanbomb/neosjs/API/CloudXInterface.js:1703:26
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
    at async CloudXInterface.GetMessages (/home/raith/test/node_modules/@bombitmanbomb/neosjs/API/CloudXInterface.js:1697:12)
    at async CloudXInterface.GetUnreadMessages (/home/raith/test/node_modules/@bombitmanbomb/neosjs/API/CloudXInterface.js:1664:12)
    at async /home/raith/test/node_modules/@bombitmanbomb/neosjs/API/MessageManager.js:74:26

--------

/home/raith/test/node_modules/@bombitmanbomb/neosjs/API/CloudXInterface.js:1589
      for (let item of b.Entity) a.Add(new Friend(item));
                         ^

TypeError: b.Entity is not iterable
    at /home/raith/test/node_modules/@bombitmanbomb/neosjs/API/CloudXInterface.js:1589:26
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
    at async CloudXInterface.GetFriends (/home/raith/test/node_modules/@bombitmanbomb/neosjs/API/CloudXInterface.js:1584:12)
    at async CloudXInterface.GetFriends (/home/raith/test/node_modules/@bombitmanbomb/neosjs/API/CloudXInterface.js:1580:14)

**Comment by bombitmanbomb** [comment]
Try loading on 1.6.10 And see if the issue persists.  The message manager had some minor changes

**Comment by RaithSphere** [comment]
okay will do, just updated and got this as i was booting up

```
/home/raith/test/node_modules/@bombitmanbomb/neosjs/Plugins/CommandHandler.js:87
                                new this.Handler(context.Neos, context, Message, Extra)
                                ^

TypeError: this.Handler is not a constructor
    at Neos.Run (/home/raith/test/node_modules/@bombitmanbomb/neosjs/Plugins/CommandHandler.js:87:5)
    at Neos.emit (node:events:339:22)
    at Events.<anonymous> (/home/raith/test/node_modules/@bombitmanbomb/neosjs/Neos.js:160:20)
    at Events.emit (node:events:327:20)
    at Neos.CloudXInterface.Messages.onMessageReceived (/home/raith/test/node_modules/@bombitmanbomb/neosjs/Neos.js:87:16)
    at /home/raith/test/node_modules/@bombitmanbomb/neosjs/API/MessageManager.js:108:44
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
    at async /home/raith/test/node_modules/@bombitmanbomb/neosjs/API/MessageManager.js:75:26
```

**Comment by bombitmanbomb** [comment]
What version of Node?

**Comment by RaithSphere** [comment]
v15.0.1 on Ubuntu 18.04.5 LTS

**Comment by bombitmanbomb** [comment]
Hmm, latest I've tested with is 14.0.0, will add a test case for 15. 

Are you invoking ghe CommandHandler to an object with New?

**Comment by RaithSphere** [comment]
This is my full code i am using right now and i am using new yes

```
const NEOS = require('@bombitmanbomb/neosjs');
const CommandHandler = require('@bombitmanbomb/neosjs/Plugins/CommandHandler');
const HeadlessInterface = require("@bombitmanbomb/neosjs/Plugins/HeadlessInterface");
const Neos = new NEOS({StatusInterval:0}); // Disable Status Updates - Can conflict with the Headless Status
const Command = new CommandHandler(Neos, ''); // Pass Neos client and [Error Message] (Optional)
Neos.on('messageReceived', Command.Run); // Route messages to the Command Handler Directly (Shorthand).

var Headless;
//Linux
Headless = new HeadlessInterface("/home/raith/.steam/SteamApps/common/NeosVR/","/home/raith/.steam/SteamApps/common/NeosVR/Config/Sweven.json") // Pass Path to Neos Folder OR child_process.spawn() of Headless

Neos.on('login', () => {
  console.log(Neos.CurrentUser); // Log the current user
  Neos.SendTextMessage("U-RaithSphere", "<B>Headless Online and waiting for commands</B>");
});

Neos.on("friendAdded",(friend)=>{
    if (friend.FriendStatus == "Requested") {
          var s = `Friends request from ${friend.FriendUsername}<BR><color=red>Type /sys acceptFriendRequest ${friend.FriendUsername}</color><BR> to accept`;
          Neos.SendTextMessage("U-RaithSphere", s);
    }
})

Neos.on("messageReceived",(message)=>{
    if(message.SenderId != "U-RaithHeadless") // Ignore headless message
	{
	   if(message.SenderId != "U-RaithSphere")
	    {
	    var m = `${message.SenderId} said ${message.Content}`;
	    Neos.SendTextMessage("U-RaithSphere", m) // Send to Raith
            }
	}
})
Command.Add('/commands', (Handler, Sender, Args) => {
  console.log(Sender + ':' + Args.join('_')); // U-RaithSphere:/commands Any_Text_After
  Handler.Reply(Object.keys(Command.Commands).sort().join('<br>')); // Reply to command in Neos a list of all Commands
});

Command.Add("/sys",(Handler, Sender, Args)=>{
  Headless.Send(Args.join(" ")).then((response)=>{
    Handler.Reply(response); 
  });
},["U-RaithSphere"]); 

//Command will invite the user to the session
Command.Add("/inviteme", async (Handler, Sender, Args)=>{
  let CloudResult = await Neos.GetUser(Sender);
  Headless.Send(`invite ${CloudResult.Content.username}`).then((response) => {
    Handler.Reply(response) // Invite Sent if Friends
  });
});
```

**Comment by bombitmanbomb** [comment]
Issue is fixed in 1.6.11, Also, Try not to have more than one messageReceived handlers.

**Comment by RaithSphere** [comment]
This seems to happen when there is an update message from Neos from what i can tell
"version": "1.6.17-rc.1"

/home/raith/test/node_modules/@bombitmanbomb/neosjs/API/CloudXInterface.js:1696
      for (let item of b.Entity) a.Add(new Message(item));
                         ^

TypeError: b.Entity is not iterable
    at /home/raith/test/node_modules/@bombitmanbomb/neosjs/API/CloudXInterface.js:1696:26
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
    at async CloudXInterface.GetMessages (/home/raith/test/node_modules/@bombitmanbomb/neosjs/API/CloudXInterface.js:1690:12)
    at async CloudXInterface.GetUnreadMessages (/home/raith/test/node_modules/@bombitmanbomb/neosjs/API/CloudXInterface.js:1657:12)
    at async /home/raith/test/node_modules/@bombitmanbomb/neosjs/API/MessageManager.js:75:26
